app/controllers/users_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      sign_in_as(@user)
+      redirect_to user_path(@user)
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @events = @user.events
+    @upcoming_events = @user.upcoming_events
+    @previous_events = @user.previous_events
```

Comment:
`[Optional]` You could use `upcoming` and `past` scopes through the `attended_events` association. I that way you don't need to repeat the queries in the User model.

Reply:
Made this change as well. :D